### PR TITLE
feat(leadspace): introduced leadspace medium variant

### PR DESCRIPTION
**Pull request by andysherman2121**
### Related Ticket(s)

#5231 

### Description

Added `medium` and `tall` `size` prop

### CODE REVIEW

**Comment by jeffchew** [comment]
@andysherman2121 looks like a bunch of build issues. Would you be able to fix today?

**Comment by andysherman2121** [comment]
> @andysherman2121 looks like a bunch of build issues. Would you be able to fix today?

@jeffchew Yup! i've been working through them

**Comment by andysherman2121** [comment]
@DragosRistici Deploy preview is up!

File: packages/web-components/src/components/leadspace/leadspace.ts [hunk 1]
```diff
@@ -59,20 +59,50 @@ class DDSLeadSpace extends StableSelectorMixin(LitElement) {
     });
   }
 
+  /**
+   * Returns a class-name based on the size parameter type
+   */
+  protected _getContainerClass() {
+    return classMap({
+      [`${prefix}--leadspace__container-medium`]: this.size === LEADSPACE_SIZE.MEDIUM,
```
Comment (asudoh) [inline packages/web-components/src/components/leadspace/leadspace.ts:67]: BEM uses two dashes for variants:

```suggestion
      [`${prefix}--leadspace__container--medium`]: this.size === LEADSPACE_SIZE.MEDIUM,
```

**Review by asudoh — COMMENTED** [review]
Thank you for the update @andysherman2121!

**Comment by jeffchew** [comment]
@andysherman2121 instead of a knob, can it be a separate story? Probably two more, with and without image.

@shixiedesign I'm not sure if we wanted to remove the small variants in this PR or not? I think I recall discussing that but couldn't find that in any issues.

**Comment by jeffchew** [comment]
The documentation for `type` is outdated, can you update to include `medium`?

Also, the stories are a bit confusing, Wondering if we should rename the stories and have the available options as `tall` (not default), `medium`, `centered` (and not sure if we're removing `small`, @shixiedesign ?)

Screen Shot 2021-03-11 at 6 38 10 PM

**Comment by andysherman2121** [comment]
> @andysherman2121 instead of a knob, can it be a separate story? Probably two more, with and without image.
> 
> @shixiedesign I'm not sure if we wanted to remove the small variants in this PR or not? I think I recall discussing that but couldn't find that in any issues.

@jeffchew James said that it should be a `knob`. I can definitely make it into a story though if that's what we want instead

**Comment by jeffchew** [comment]
> @jeffchew James said that it should be a `knob`. I can definitely make it into a story though if that's what we want instead

I think we should, but see my comment above (I changed my comment a bit). My goal is more about trying to capture the visual regression coverage with Percy.

**Comment by shixiedesign** [comment]
Until @DragosRistici wakes up, here's a few things I noticed: 
- [ ] Buttons & copy are not lower left pinned. 
Currently: 
image
Expecting:
image

- [ ] Probably related to above, when `sm` breakpoint and line wraps, buttons gets pushed out of lead space. Expecting buttons and small copy to be bottom pinned 
Mar-11-2021 22-03-09

**Comment by andysherman2121** [comment]
@shixiedesign Ah you're right i'll fix that

**Comment by andysherman2121** [comment]
@shixiedesign Updated! The copy and buttons should be pinned to the bottom left now

**Comment by DragosRistici** [comment]
First of all, thank you so much, Andy @andysherman2121 , for your fast turnaround. And thank you to Shixie @shixiedesign  for getting to it first.


I’d like to start with a couple of small general tweaks.  

1. Because we are showcasing a white lead space (no image) on a white background page, I suggest adding a Gray10 background color to the bx--leadspace__overlay div, so the adopter can “see” the lead space, as we are giving him height options with quite significant differences. Or at least add a border-bottom of 1px solid gray30. (Of course, if the gradient wouldn't be there when there's no image, that would be awesome, but i think its fine with the gray10 behind it.)
Screenshot 2021-03-12 at 15 19 19


2. As the design specs say, we should use this opportunity to showcase it as close as possible to the real world, that’s why we added the text “Heading could go on two lines max” because headings usually go on more than one row. 



///////////////

DESKTOP BREAK POINTS (max, xlg, lg)


3. Display-01’ has a bottom spacing of 6rem, like in the image below. And I would like to use this occasion to ask you guys to use margin to set the distance between elements, because if you use padding, you’ll just make the element bigger and the padding space will get the styling if case may be, too.    It could be like this:
Screenshot 2021-03-12 at 13 22 48

because the design specs are like this:
image




4. Expressive-heading-03 should have no top spacing, but a bottom spacing of 2rem;  


5. Button group should have no extra spacing top/bottom. But the Dds-button-group-item has a margin-top of 1rem. If the component has it already, then maybe you cut it from the above element, so in the end we get A TOTAL OF 2rem between the expressive-heading-03 and the dds-button-group element.  

Attention: Please, make sure that the margin-top of 1 rem of the button item we have does not get used in other components like that. If that’s the case, then for the lead space reduce it from the above element.
image




///////////////

TABLET BREAK POINT (md)

Screenshot 2021-03-12 at 14 46 43Screenshot 2021-03-12 at 14 47 01


Please follow the specs for this one. The image is above the rest, just like the mobile version. If you need to see another example, please check the ibm.com homepage as they are using the same visual approach for the md break point. 

Screenshot 2021-03-12 at 15 14 05Screenshot 2021-03-12 at 15 42 20Screenshot 2021-03-12 at 15 13 58
Screenshot 2021-03-12 at 15 13 42



///////////////

MOBILE BREAK POINT (sm)

Also, for the mobile, please take the same approach. There is no lead space height for the md and sm break points. Here the elements have precise spacing between them, exactly like in the specs. If there is an image, it will not be as background. 

Screenshot 2021-03-12 at 15 10 08Screenshot 2021-03-12 at 15 50 43
Screenshot 2021-03-12 at 15 09 15

Thank you so much for your effort and let me know if you have any questions. Thanks.

**Comment by andysherman2121** [comment]
@DragosRistici Thanks for the comments! I talked to @photodow and he mentioned that for this story we weren't including the image functionality?

**Comment by DragosRistici** [comment]
@andysherman2121 I didn't know that, but if he says so, then it must be true. One last thing, please, about the gradient. Look at the specs, there are 3 points of contact for it: 0, 1/2 and 3/4. They have the opacity set to 1, .4 and 0, respectively. Right now, there are 2, and one is .95 and the other one is 0 at 3/4. Thanks again, Andy. 


Screenshot 2021-03-12 at 16 12 00


Screenshot 2021-03-12 at 16 09 26
Screenshot 2021-03-12 at 16 09 21
Screenshot 2021-03-12 at 16 09 07

**Comment by andysherman2121** [comment]
Hi @DragosRistici !  

I've added the `gray-10` background and fixed the gradient. I couldn't get rid of the spacing above the button-group so I took  your advice and changed the bottom spacing on the description text.

**Comment by DragosRistici** [comment]
@andysherman2121 Thank you, Andy. You are a master! I rest my case. Its fine by me. Now what? :)

**Review by DragosRistici — APPROVED** [review]
Thank you.

**Review by IgnacioBecerra — APPROVED** [review]
LGTM, good job!

**Pull request merged**